src/node/blockstorage.cpp

```diff
@@ -260,8 +260,13 @@ bool BlockManager::LoadBlockIndex()
     std::sort(vSortedByHeight.begin(), vSortedByHeight.end(),
               CBlockIndexHeightOnlyComparator());
 
+    CBlockIndex* previous_index{nullptr};
     for (CBlockIndex* pindex : vSortedByHeight) {
         if (ShutdownRequested()) return false;
+        if (previous_index && pindex->nHeight > previous_index->nHeight + 1) {
+            return error("%s: block index is non-contiguous, index of height %d missing", __func__, previous_index->nHeight + 1);
```

Comment:
> The diff below passes.

Where does it pass for you?
I tried to apply the diff too, and for me it passes on this branch and fails after removing 04d533781002be054e6f2493f3ab2389135ceec5 (not because the log message isn't found, but because the bitcoin core crashe with the assert before that).
That is what I would have expected, since the diff adds an additional check for the added log "ERROR: LoadBlockIndex: block index is non-contiguous", which is supposed to be hit as a result of the perturbation of the block files.

Reply:
nit (coming from a non-native speaker): isn't non-continuous the more fitting description (meaning "(un)interrupted sequence") rather than non-contiguous (meaning "(not) sharing a common boundary")?

Reply:
Not a native speaker either, but for me "contiguous" has more of a spatial connotation (like blocks building a chain by being attached to each other), whereas "continuous" has a more temporal connotation.